### PR TITLE
feat(curriculum): javascript regular expressions quiz

### LINKED ISSUE #56209: [HACKTOBERFEST 2024] - Regular Expressions Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Regular Expressions quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-regular-expressions`.

**All quiz questions need to relate to one of these topics:**

- Regular Expression syntax (/pattern/ and RegExp())
- test()
- match()
- replace()
- matchAll()
- replaceAll()
- character classes \d \s etc
- lookaheads and lookbehinds
- quantifiers
- capture groups and backreferences

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by EggSaled**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56209

### CODE REVIEW

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Just looking at this, there's a lot of cases where you're using backticks around a type that you shouldn't be.

"X returns a `String`" - not really, `String` is the global constructor. "X returns a string".

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-regular-expressions/66edd3011f18f4ee1bd9d28b.md [hunk 1]
```diff
@@ -17,439 +17,464 @@ Answer all of the questions below correctly to pass the quiz.
...
+```js
+foo.replaceAll(/foobar/g, "fizzbuzz")
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following character classes is equivalent to the regular expression `/[a-zA-Z0-9]/`?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-regular-expressions/66edd3011f18f4ee1bd9d28b.md:220]:
```suggestion
Which of the following character classes is equivalent to the regular expression `/[a-zA-Z0-9_]/`?
```

`\w` includes the underscore.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-regular-expressions/66edd3011f18f4ee1bd9d28b.md [hunk 2]
```diff
@@ -17,439 +17,464 @@ Answer all of the questions below correctly to pass the quiz.
...
-Placeholder answer
+The backreference includes the result from a capture group as part of a pattern/sub-pattern to match.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which part of the following regular expression causes errors?
+`/(?:Expense|Asset) \$\d+\.\d{2} \1 \$\d+\.\d{2}`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-regular-expressions/66edd3011f18f4ee1bd9d28b.md:463]:
````suggestion
Which part of the following regular expression causes errors?

```js
/(?:Expense|Asset) \$\d+\.\d{2} \1 \$\d+\.\d{2}
```
````

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-regular-expressions/66edd3011f18f4ee1bd9d28b.md [hunk 3]
```diff
@@ -17,439 +17,467 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+An object containing information about the (sub)string the regular expression matched.
 
 ---
 
-Placeholder distractor 3
+null, the method `test` only validates if the given regular expression is valid.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-regular-expressions/66edd3011f18f4ee1bd9d28b.md:92]:
```suggestion
Null, the method `test` only validates if the given regular expression is valid.
```

Or wrap it in backticks as `` `null` `` if referring to the specific value.

**Review by naomi-lgbt — COMMENTED** [review]
Last nit.

**Review by naomi-lgbt — APPROVED** [review]

**Review by gikf — COMMENTED** [review]
Two nitpicks

**Review by gikf — APPROVED** [review]

**Pull request merged**